module1/src/test/java/net/broscorp/primitives/PrimitivesTest.java

```diff
@@ -0,0 +1,50 @@
+package net.broscorp.primitives;
+
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+public class PrimitivesTest {
+
+  @Test
+  public void overflowTest() {
+
+    //variable is in positive range
+    int testedVariable = 1;
+    //adding positive number
+    testedVariable += Integer.MAX_VALUE;
+
+    //for some reason variable is now negative(((
+    assertTrue(testedVariable < 0);
+
+  }
+
+  @Test
+  public void failCastTest() {
+    //long max value is far more than int max value
+    //this operation results in overflow
+    long extremelyBigLong = Long.MAX_VALUE;
+    assertNotEquals(extremelyBigLong, (int) extremelyBigLong);
+
```

Comment:
What will happen if we cast a narrower integer type to a broader integer type?

Reply:
Casting narrower types to broader is considered a safe cast and will always be casted correctly. Implicit casting is performed internally by compiler, explicit is forced via relevant synaxis.  Explicit is used when the user is confident in the safety of cast and for some reason needs a different type.